truncating the sequence also when the input is already tokenised

lfoppiano opened the pull request:
See comments in #44 

I had to park the change somewhere. Feel free to ignore it if incorrect.

kermitt2 commented:
So you also need to truncate `batch_y` up to `max_length_x` otherwise the sequence size for labels won't match the sequence size of the tokens. I think just adding on line 112:

```
if self.y is not None:
            # note: tags are always already "tokenized",
            batch_y = np.zeros((max_iter, max_length_y), dtype='float32')
            batch_y = [ row[:max_length_y] for row in batch_y ]  # added line 112
```

(no need to recalculate `max_length_y`, it is the same as `max_length_x`)

lfoppiano commented:
I fixed the code based on your input and tried to simplify the various implementation, might require some more check because I'm not sure i've introduced more bugs...

kermitt2 commented:
I've done some tests with or without ELMo, BERT, ... and ensured that batch_y is truncated too (as in Daniel version). 
Thanks !
